src/libcore/iter/iterator.rs

```diff
@@ -1514,6 +1514,24 @@ pub trait Iterator {
         false
     }
 
+    /// Completly consumes an iterator. This is a conveniant method for
+    /// functional programming with iterators.
+    ///
+    /// ```
+    /// let xs = [0, 1, 2, 3];
+    /// let mut sum = 0;
+    ///
+    /// xs.iter().map(|x| sum += x).drain();
+    /// assert_eq!(6, sum);
+    /// ```
+    #[inline]
+    #[stable(feature = "iterator_helper", since = "1.22.0")]
```

Comment:
It's a minor thing, but to keep it as predictable as possible, *iter_methodname* or *iterator_methodname* would be a good feature name. Depends on what the method name will be, of course, just change it with the method name.